### PR TITLE
Fix issue #12000: Parsing arXiv Id when importing a PDF with arXiv Id

### LINKED ISSUE #12000: Importing a PDF with arXiv Id should fetch the arXiV information using the arXivFetcher
Example paper: https://arxiv.org/abs/2408.06224

Image

Similar to `doi = getDoi(null);` and related code in org.jabref.logic.importer.fileformat.PdfContentImporter, there should be a handling of the àrXiv` id.

---

Tasks:

1. Add test case to `org.jabref.logic.importer.fileformat.PdfContentImporterTest`
2. Implement functionality in `org.jabref.logic.importer.fileformat.PdfContentImporter` (and maybe also the classes for the arXiv Id parsing)

### PR DESCRIPTION
**Pull request by XYZ567AB**
Describe the changes you have made here: what, why, ...
Link the issue that will be closed, e.g., "Closes #333". If your PR closes a koppor issue, link it using its URL, e.g., "Closes https://github.com/koppor/jabref/issues/47".

Add 'ARXIVID' field to 'StandardField' enum class
Add 'getArxivId' method to 'PdfContentImporter' class
Add logic to handle arxivId in 'getEntryFromPDFContent' method
Add 'extractArxivIdFromPage1' test method to 'PdfContentImporterTest' class

Closes #12000
Closes https://github.com/JabRef/jabref/issues/12000

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [X] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [X] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [X] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [X] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/entry/field/StandardField.java [hunk 1]
```diff
@@ -25,6 +25,7 @@ public enum StandardField implements Field {
     ARCHIVEPREFIX("archiveprefix"),
     ASSIGNEE("assignee", FieldProperty.PERSON_NAMES),
     AUTHOR("author", FieldProperty.PERSON_NAMES),
+    ARXIVID("arXivId", FieldProperty.VERBATIM),
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/model/entry/field/StandardField.java:28]: Arxiv is no StandardField, you need to use eprint for this
see chapter 3.14.7 Electronic Publishing Information of the biblatex spec
 http://mirrors.ctan.org/macros/latex/contrib/biblatex/doc/biblatex.pdf

File: src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java [hunk 2]
```diff
@@ -123,4 +123,38 @@ British Journal of Nutrition (2008), 99, 1–11 doi: 10.1017/S0007114507795296
...
     }
+
+    @Test
+    void extractArxivIdFromPage1() {
+        BibEntry entry = new BibEntry(StandardEntryType.InProceedings)
+                .withField(StandardField.DOI, "10.1017/S0007114507795296")
+                .withField(StandardField.AUTHOR, "Review Article")
+                .withField(StandardField.TITLE, "British Journal of Nutrition (2008), 99, 1–11 doi: 10.1017/S0007114507795296 arXiv:2408.06224v1 q The Authors")
+                .withField(StandardField.YEAR, "2024")
+                .withField(StandardField.ARXIVID, "2408.06224v1");
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java:134]: Needs to be eprint

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
Arxiv is not a Standard field

File: src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java (hunk 2 above)
Comment (XYZ567AB) [reply src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java:134]: Thanks for your feedback! I see, I will change it.

**Comment by koppor** [comment]
@XYZ567AB You can add a code comment into StandardField.java to guide someone also working here. Saying how the arXiv ID is stored and also add a link to the Java class handling arXiv Ids and the functionality introduced at https://github.com/JabRef/jabref/pull/11627.

File: src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java (hunk 2 above)
Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java]: You then should also add the correpsonding eprinttype field to arxiv

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 
 ### Added
 
+- We added functionality to handle arXiv ID in `PdfContentImporter` and implemented related test case. [#12000](https://github.com/JabRef/jabref/issues/12000)
```
Comment (koppor) [inline CHANGELOG.md:14]: These comments are user facing. A user does not see `PdfContentImporter` in the UI. We are updating the docs at https://github.com/JabRef/user-documentation/pull/537 - and the updated section should be linked.

File: src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java [hunk 4]
```diff
@@ -480,6 +490,21 @@ private String getDoi(String doi) {
...
+            if (pos < 0) {
+                pos = curString.indexOf("arXiv");
+            }
+            if (pos >= 0) {
+                String arxivText = curString.substring(pos);
+                return ArXivIdentifier.parse(arxivText).map(ArXivIdentifier::asString).orElse(null);
+            }
+        }
+        return arxivId;
+    }
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java:506]: No! We have the class org.jabref.model.entry.identifier.ArXivIdentifier use this.

File: src/main/java/org/jabref/model/entry/identifier/ArXivIdentifier.java [hunk 5]
```diff
@@ -45,13 +45,13 @@ public static Optional<ArXivIdentifier> parse(String value) {
...
-            return getArXivIdentifier(oldIdentifierMatcher);
-        }
+       Pattern oldIdentifierPattern = Pattern.compile("(" + ARXIV_PREFIX + ")?\\s?:?\\s?(?<id>(?<classification>[a-z\\-]+(\\.[A-Z]{2})?)/\\d{7})(v(?<version>\\d+))?");
+       Matcher oldIdentifierMatcher = oldIdentifierPattern.matcher(identifier);
+       if (oldIdentifierMatcher.matches()) {
+           return getArXivIdentifier(oldIdentifierMatcher);
+       }
 
-        return Optional.empty();
+       return Optional.empty();
```
Comment (koppor) [inline src/main/java/org/jabref/model/entry/identifier/ArXivIdentifier.java:54]: Wrong indent. Please revert.

File: src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java [hunk 6]
```diff
@@ -123,4 +123,38 @@ British Journal of Nutrition (2008), 99, 1–11 doi: 10.1017/S0007114507795296
 
         assertEquals(Optional.of(entry), importer.getEntryFromPDFContent(firstPageContent, "\n"));
     }
+
+    @Test
+    void extractArxivIdFromPage1() {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/importer/fileformat/PdfContentImporterTest.java:128]: Please also use a real arXiv PDF. I think, there is a link to one in the issue?

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java [hunk 7]
```diff
@@ -458,6 +465,9 @@ Optional<BibEntry> getEntryFromPDFContent(String firstpageContents, String lineS
             entry.setField(StandardField.PAGES, pages);
         }
         if (year != null) {
+            if (arxivId != null) {
+                year = "20" + arxivId.substring(0, 2);
+            }
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java:470]: This should go to the `else` branch. - If there is no year then one can check for `arxivId` and guess a year out of it.

File: src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java (hunk 4 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/importer/fileformat/PdfContentImporter.java:506]: This is still unfixed.

**Review by koppor — CHANGES_REQUESTED** [review]

**Comment by koppor** [comment]
Closing this issue due to inactivity :zzz:

**Pull request closed without merging**